workflow/controller/dag.go

```diff
@@ -44,7 +44,7 @@ type dagContext struct {
 
 	// dependencies is a list of all the tasks a specific task depends on. Because dependencies are computed using regex
 	// and regex is expensive, we cache the results so that they are only computed once per operation
-	dependencies map[string][]string
+	dependencies map[string]map[string]common.DependencyType
```

Comment:
I'm almost surely wrong, but I didn't see a place where we actually use the value of the inner map (i.e. `common.DependencyType`). Can you explain why it's necessary, point me where it's used?

Reply:
The issue is that dagValidationContext needs the inner map, and was getting passed in as a dagContext, so I modified dagContext to keep them matching. This wasn't ideal, I've fixed it up.